### DISCUSSION TITLE
Tasmota devices reaching out to public URLs -clarification request

### DISCUSSION THREAD

**Discussion by titiviking** [arendst/Tasmota#20565] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
Tasmota-based devices seem to call out to the internet, while they are meant to be local-only.

List of Ip's (and reverse DNS) of connections blocked by my firewall:

- 108.61.189.74 (mail.emuadmin.com)
- 5.255.124.190
- 185.224.145.68 (ntp.emenders.nl)
- 162.159.200.123 (time.cloudflare.com)
- 185.153.41.4 (ntp1.rtbf.be)
- 47.91.73.128 
- 54.194.162.25 (ec2-54-194-162-25.eu-west-1.compute.amazonaws.com)
- 94.198.159.11 (nts1.time.nl)
- 45.87.78.35 
- 54.193.41.31 (ec2-54-193-41-31.us-west-1.compute.amazonaws.com)

Can the use of these be clarified so appropriate firewall rules can be configured if necessary ?

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [x] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 10.0
  - [x] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**
[ntp.emenders.nl](https://mxtoolbox.com/SuperTool.aspx?action=ptr%3a5.255.124.190&run=toolpage#)

**Comment by s-hadinger** [comment]
The only servers being connected are NTP servers. Nothing more. The three domains used are `2.pool.ntp.org`, `2.europe.pool.ntp.org` and `2.nl.pool.ntp.org`.

When you do a firmware update originated by Tasmota, the domain used is `ota.tasmota.com` which uses AWS CloudFront as a CDN, but you can set the URL to whatever you want.

**Reply by sfromis** [reply]
Outside of NTP, the cases for standard Tasmota using a public URL is doing OTA upgrades, or a couple of specific ESP32 tools, but only if you click on the features. And the time servers posted are not configured into Tasmota.

What you have is not coming from Tasmota as such. If it is actually coming from a Tasmota device, it is possible that someone already having gotten into your network might have "upgraded" to a binary including "additional unwanted features".

**Comment by Jason2866** [comment]
If you provide a NTP server in your LAN there is no outgoing traffic at all.

**Reply by sfromis** [reply]
You could even have a Tasmota device with a RTC or GPS chip act as a local NTP server.

**Comment by barbudor** [comment, marked as answer]
`2.pool.ntp.org`, `2.europe.pool.ntp.org` and `2.nl.pool.ntp.org` are virtual using a load balancing managed by ntp.org. Depending on your location or time of the day, resolution to an IP address could be different and variable

So any of those IP addresses you are listing could be the ones behind the NTP servers used by Tasmota, and they could change at each sync.

From the names I could guess that you could be located in Belgium or Netherlands
Being located in France, I get totally different IPSs for those servers

As pointed by Jason, forcing to use a local NTP server (most routers can act as a NTP server) would totally avoid those.

**Comment by titiviking** [comment]
I had some network changes and my first NTP server was not available anymore, hence it falled back onto secondary NTP server...
Restoring a working local NTP server, I do no see the firewall events anymore, so problem solved. Thanks for the clarifications
